ttyclock.c

```diff
@@ -686,4 +689,80 @@ main(int argc, char **argv)
      return 0;
 }
 
+time_t timestamp(void)
+{
+     if (ttyclock.rounding_permitted && whole_seconds_delay())
+     {
+          return rounded_timestamp();
+     }
+     else
+     {
+          return time(NULL);
+     }
+}
+
+time_t rounded_timestamp(void)
+{
+     static const long HALF_SECOND_IN_NANOSECONDS = 500 * 1000 * 1000;
+     struct timespec currentTime;
+
+     if (clock_gettime(CLOCK_REALTIME, &currentTime) != 0)
+     {
+          return time(NULL);
+     }
+
+     if (currentTime.tv_nsec >= HALF_SECOND_IN_NANOSECONDS)
+     {
+          return currentTime.tv_sec + 1;
+     }
+     else
+     {
+          return currentTime.tv_sec;
+     }
+}
+
+struct timespec sleep_length(void)
+{
+     static const long SECOND_IN_NANOSECONDS = 1000 * 1000 * 1000;
+     static const long NINE_TENTHS = 900 * 1000 * 1000;
+
+     if (whole_seconds_delay())
+     {
+          struct timespec currentTime;
+
+          if (clock_gettime(CLOCK_REALTIME, &currentTime) == 0 &&
+               currentTime.tv_nsec > 0)
+          {
+               struct timespec result;
+
+               if (currentTime.tv_nsec < NINE_TENTHS) {
+                    result.tv_sec = ttyclock.option.delay - 1;
+               }
+               else
+               {
+                    result.tv_sec = ttyclock.option.delay;
+               }
+
+               result.tv_nsec = SECOND_IN_NANOSECONDS - currentTime.tv_nsec;
+               return result;
+          }
+     }
```

Comment:
this magic stuff could use some inline comments. why do we care about NINE_TENTHS?

Reply:
Will comment